matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/MatchingNotificationResource.java

```diff
@@ -175,4 +175,22 @@ Response getEmailToBeSent(@FormParam("matchId") String matchId,
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/set-status")
     Response setStatus(@FormParam("matchesIds") Set<Long> matchesIds, @FormParam("status") String status);
+
+    /**
+     * Returns a JSON object containing all matches where reference/matched patient ID is same as one of parameters.
+     *
+     * @param referencePatientId id of reference patient to load matches for
+     * @param referenceServerId id of the server that hosts reference patient
+     * @param matchedPatientId id of matched patient to load matches for
+     * @param matchedServerId id of the server that hosts matched patient
+     * @return a single matchID
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/get-matchid")
+    Response getMatches(@FormParam("referencePatientId") String referencePatientId,
```

Comment:
the method is named `getMatches()`, but URL is `get-matchid`, and it actually returns only match ID, not a list of matches (so the comment is incorrect)